app/controllers/questions_controller.rb

```diff
@@ -32,6 +32,30 @@ def create
 
     respond_to do |format|
       if @question.save
+        # check if there are any tags for the question
+        if params[:tags_list]
+          # get names of tags the user wants to add to the question
+          tag_names_array = params[:tags_list].split(",")
+          # tags[] = array for storing Tag instances associated with the names in tag_names_array
+          tags = []
+          # loop through names from user input.
+          tag_names_array.each do |tagName|
+            # check if tag with this tagName already exists
+            if Tag.exists?(tag: tagName)
+              # get the Tag that has this tagName
+              @t = Tag.where(tag: tagName).first
+            else
+              # make a new Tag with this tagName
+              @t = Tag.new(:tag => tagName)
+              @t.save
+            end
+            # add Tag to tags[]
+            tags.push(@t)
+          end
+          # set the question's tags to tags[]
+          # this allows for both creation and deletion of QuestionTagJunction instances?
+          @question.tags = tags
```

Comment:
Yes, because @question isn't actually changed by adding tag associations. The only things that change are Tags and QuestionToTagJunctions.